examples/megacli.cpp

```diff
@@ -1659,11 +1727,17 @@ static void setprompt(prompttype p)
     else
     {
         pw_buf_pos = 0;
+#if defined(WIN32) && defined(NO_READLINE)
+        static_cast<WinConsole*>(console)->updateInputPrompt(prompts[p]);
+#else
         cout << prompts[p] << flush;
+#endif
         console->setecho(false);
     }
 }
 
+
+#ifdef THIS_CODE_EXISTS_IN_MEGA_LIB
```

Comment:
thanks for identifying the problem - I wasn't sure if the MEGAAPI_H would be a reliable indicator, so I've just chosen a slightly different class name to avoid the link error where megaapi_impl is included in the library build.  The compiled executable will turn out the same.